Improves and fixes inconsistencies in boundary conditions docs (#1836)
CliMA/Oceananigans.jl: docs/src/model_setup/boundary_conditions.md

    * "No flux" or "free slip" boundary conditions (default for tracers and wall-tangential momentum components in `Bounded` directions).

3. `ValueBoundaryCondition` (aka "Dirichlet" boundary conditions) models boundary fluxes given a field's diffusive flux model, and assuming that a field has the prescribed value on the boundary.
   _Note_: You cannot use `ValueBoundaryCondition` on a wall-normal velocity component; you must use `Open` for that.
   Examples where you might use `ValueBoundaryCondition`:
   * Prescribe a surface to have a constant temperature, like 20 degrees.
     Heat will then flux in and out of the domain depending on the temperature difference between the surface and the interior, and the temperature diffusivity.
   * Prescribe a velocity tangent to a boundary as in a driven-cavity flow (for example), where the top boundary is moving.
     Momentum will flux into the domain do the difference between the top boundary velocity and the interior velocity, and the prescribed viscosity.

4. `FluxBoundaryCondition` _directly_ prescribes the flux of a quantity across a boundary rather than calculating it given a viscosity or diffusivity.

>    For example, sunlight absorbed at the ocean surface imparts a temperature flux that heats near surface fluid.

Reviewer: ```suggestion
   For example, sunlight absorbed at the ocean surface imparts a temperature flux that heats near-surface fluid.
```

>    If you have a known diffusivity, you can express `FluxBoundaryCondition(flux)` using `GradientBoundaryCondition(-flux / diffusivity)` (aka "Neumann" boundary condition).

Reviewer: ```suggestion
   If there is a known `diffusivity`, you can express `FluxBoundaryCondition(flux)` using `GradientBoundaryCondition(-flux / diffusivity)` (aka "Neumann" boundary condition).
```
